### PR TITLE
OLS-147: Create a skeleton OLS operator

### PR DESCRIPTION
**Pull request by vbelouso**
## Description

Added initial version of operator.

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Closes https://issues.redhat.com/browse/OLS-147

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Adds namespace to all resources.
+namespace: lightspeed-operator
```
Comment (bparees) [inline config/default/kustomization.yaml:2]: i think we will want this to be `openshift-lightspeed-operator`, to distinguish from ansible lightspeed.  Plus being in an openshift prefixed namespace allows our metrics to be reported to telemetry

File: PROJECT [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: openshift.io
+layout:
+- go.kubebuilder.io/v4
+plugins:
+  manifests.sdk.operatorframework.io/v2: {}
+  scorecard.sdk.operatorframework.io/v2: {}
+projectName: lightspeed-operator
```
Comment (bparees) [inline PROJECT:11]:
```suggestion
projectName: openshift-lightspeed-operator
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Build the manager binary
+FROM golang:1.20 as builder
```
Comment (raptorsun) [inline Dockerfile:2]: We can use the latest `golang:1.21` as builder image.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# IMAGE_TAG_BASE defines the docker.io namespace and part of the image name for remote images.
+# This variable is used to construct full image tags for bundle and catalog images.
+#
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# quay.io/openshift/lightspeed-operator-bundle:$VERSION and quay.io/openshift/lightspeed-operator-catalog:$VERSION.
+IMAGE_TAG_BASE ?= quay.io/vbelouso/lightspeed-operator
```
Comment (raptorsun) [inline Makefile:32]: Shall we add a `#todo` to remind us to change the repository to that under openshift organization after our prow job pushing image to openshift/lightspeed-operator repository is ready?

File: api/v1alpha1/olsconfig_types.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+type ModelSpec struct {
+	// Model name
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// Model API URL
+	URL string `json:"url,omitempty"`
+}
+
+// ModelSpec defines the desired state of LLM provider.
```
Comment (raptorsun) [inline api/v1alpha1/olsconfig_types.go:144]: typo: `s/ModelSpec/ProviderSpec/`

**Review by raptorsun — APPROVED** [review]
/lgtm
The skeleton of the operator looks good to me 👍 
If @xrajesh agrees, let's merge it and move to next step: setup pre-merge tests using empty/dummy tests.

**Pull request merged**